scripts/dep-supply-chain-check.py

```
@@ -0,0 +1,956 @@
+    for group in (proj.get("optional-dependencies") or {}).values():
+        specs.extend(group or [])
+    names = []
+    for d in specs:
+        m = re.match(r"^[A-Za-z0-9._\-]+", str(d).strip())
+        if m:
+            names.append(unify_pypi(m.group(0)))
+    return names
+
+
+def parse_gosum(text):
+    """go.sum → ({module: version}, {(module, version)} 普通/​go.mod 双行集合)"""
```

Ruff (0.16.1)

[error] Invalid unescaped character zero-width-space, use "\u200B" instead

Replace with escape sequence

(PLE2515)